### PR TITLE
Update layout of settings schema

### PR DESCRIPTION
**Pull request by grzegorz914**
Hi @merdok 
this PR update layout of config.schema which make the settings layout more clear and comfortable to use.

Thanks
Grzegorz

### CODE REVIEW

**Comment by merdok** [comment]
Hi,

thanks for the PR. This looks indeed much more cleaner and better. But there are some issues:
- none of the buttons does not work, the values are not written to the config
- if you press Enter while adding a button in any of the inputs it will add endless "Name" fields.

Could you have a look at that?
You can validate the config easily here: https://hamidihamza.com/ajsf/?framework=bootstrap-4

**Comment by grzegorz914** [comment]
All fixed.

File: config.schema.json [hunk 1]
```diff
@@ -5,302 +5,301 @@
...
+                        "type": "string",
+                        "default": "My LG TV",
+                        "minLength": 1,
+                        "required": true
+                    },
+                    "ip": {
+                        "title": "TV ip address",
+                        "description": "The ip address of your TV.",
+                        "type": "string",
+                        "default": "192.168.1.5",
```
Comment (merdok) [inline config.schema.json:27]: Please no default value here

File: config.schema.json [hunk 2]
```diff
@@ -5,302 +5,301 @@
...
+                        "format": "ipv4",
+                        "required": true
+                    },
+                    "mac": {
+                        "title": "TV mac address",
+                        "description": "The mac address of your TV.",
+                        "default": "ab:cd:ef:fe:dc:ba",
```
Comment (merdok) [inline config.schema.json:34]: Here also please no default value here

**Comment by merdok** [comment]
This looks already much better! Thanks! What i noticed, when i open my existing config then always only the first appButton or notificanButton etc is shown i have to press on the "Add ..." button so that the other entires will display. Not sure if this was like that?

**Comment by grzegorz914** [comment]
Ok,

will check this again.

File: config.schema.json (hunk 2 above)
Comment (grzegorz914) [reply config.schema.json]: Fixed

File: config.schema.json (hunk 1 above)
Comment (grzegorz914) [reply config.schema.json]: Fixed

**Comment by grzegorz914** [comment]
I don’t know why the array are not automatically display all configured buttons. I have tested this on my own plugins and sometimes was display all sometimes only first. May be something wrong with Config UI X.
The layout is correct configured and write and save all value to the config.json.

**Comment by merdok** [comment]
I really like how compact and clean the config now is. Very good job on that! But we need to figure out why only the first entries are present in arrays. I noticed basically every array is now affected of this. This might confuse people. I wonder if there are any

**Comment by grzegorz914** [comment]
Yes, it is very strange, I have already tested same Schema with my denon plugin and here the array with inputs automatically displays all values. I workin on it, be patient😀

**Comment by merdok** [comment]
index.js has nothing to do with config.schema.json, those files are completely separate and are not in any way related. But i am also not sure what causes the issue. I do not know if this is a bug in the config.schema.json or maybe it is a bug in homebridge-config-ui-x. I also could not find any examples which use such complex properties defines as in this plugin. We need to investigate this further and find a fix, since i really like the new layout 👍

**Comment by grzegorz914** [comment]
I think it is not a bug in Config-UI-X because my homebridge-denon-tv plugin uses same schema and there the Inputs array are displayed all values after expand it. I will fight with this problem, this must be working. :)

**Comment by merdok** [comment]
That is true, if you say that it is working on another plugin of yours then it also should work here. Maybe there is some little thing which you are missing?

**Comment by merdok** [comment]
Any news on this?

**Comment by grzegorz914** [comment]
Nothing new, this must be issue with Config UI X. I have report this as bug in Config UI X repo and waiting on answer.

**Comment by merdok** [comment]
Ok...

**Comment by merdok** [comment]
Any news on this?

**Pull request closed without merging**